TriblerGUI/widgets/channelpage.py

```diff
@@ -72,7 +72,20 @@ def update_result_list(self):
             self.window().channel_torrents_list.set_data_items(self.playlists + self.torrents)
 
     def received_torrents_in_channel(self, results):
-        for result in results['torrents']:
+        def sort_key(torrent):
+            """ Scoring algorithm for sorting the torrent to show liveness. The score is basically the sum of number
+                of seeders and leechers. If swarm info is unknown, we give unknown seeder and leecher as 0.5 & 0.4 so
+                that the sum is less than 1 and higher than zero. This means unknown torrents will have higher score
+                than dead torrent with no seeders and leechers and lower score than any barely alive torrent with a
+                single seeder or leecher.
+            """
+            seeder_score = torrent['num_seeders'] if torrent['num_seeders'] or torrent['last_tracker_check'] > 0\
+                else 0.5
+            leecher_score = torrent['num_leechers'] if torrent['num_leechers'] or torrent['last_tracker_check'] > 0\
+                else 0.4
+            return seeder_score + leecher_score
```

Comment:
Could you turn this into `seeder_score + .5 * leecher_score`: that is more in line with the expected download speed (based on homogenous bandwidth and uniformly distributed leecher file completion)?